### PR TITLE
feat: update fetch and add tests

### LINKED ISSUE #45: Write unit tests for lib/fetch

### PR DESCRIPTION
**Pull request by qgolsteyn**
This PR updates fetch function docs, and adds some initial tests running against actual URLs to verify that fetching works correctly, both locally and on CI.

Closes #45

### CODE REVIEW

File: lib/fetch.js [hunk 1]
```diff
@@ -1,158 +1,185 @@
...
+ * @param {*} date the date associated with this resource, or false if a timeseries data
+ * @param {*} alwaysRun fetches from URL even if resource is in cache, defaults to false
+ */
+export const csv = async (url, date, alwaysRun = false) => {
   return new Promise(async (resolve, reject) => {
-    const body = await fetch(url, 'csv', date);
+    const body = await fetch(url, 'csv', date, alwaysRun);
 
-    if (!body) {
+    if (body === undefined) {
```
Comment (lazd) [inline lib/fetch.js]: Hmmmm, I do we want to try to parse empty string and `null`? Same thing above too.

File: lib/fetch.js [hunk 2]
```diff
@@ -1,158 +1,185 @@
...
+    console.log('  ❌ Got error %d trying to fetch %s headless', response._status, url);
+    browser.close();
+    return null;
+  } catch (err) {
     browser.close();
-  }
 
-  const $ = await cheerio.load(html);
+    if (err.name === 'TimeoutError') {
+      console.log('  ❌ Timed out trying to fetch %s headless', url);
```
Comment (lazd) [inline lib/fetch.js]: Future: capture timeouts in error report? Right now, it just fails.

File: lib/fetch.js [hunk 3]
```diff
@@ -1,158 +1,185 @@
...
-      return null;
+      return html;
     }
-  } else if (alwaysRun) {
-    const browser = await puppeteer.launch();
-    const page = await browser.newPage();
-    page.setContent(html.toString());
-    await page.waitFor(3000);
-    html = await page.content();
+    console.log('  ❌ Got error %d trying to fetch %s headless', response._status, url);
```
Comment (lazd) [inline lib/fetch.js]: Future: capture in error report?

Comment (qgolsteyn) [reply lib/fetch.js]: #151 added to track

**Pull request merged**